### ISSUE TITLE
Handle incomplete Tidal releases

### ISSUE THREAD

**Issue by kellnerd** [kellnerd/harmony#30]
For some releases (pre-releases?), Tidal's API does not return all tracks:
-  https://listen.tidal.com/album/365033100 ([Cached](https://harmony.pulsewidth.org.uk/release?tidal=365033100&region=GB&ts=1718104580))
- https://listen.tidal.com/album/355013613 ([Cached](http://localhost:8000/release?tidal=355013613&region=GB&ts=1718139754))

The missing tracks are not shown on tidal.com/browse/album pages at all, on listen.tidal.com pages they are displayed greyed out.

Since the API returns at least the correct track count we could try to fill the tracklist (for single medium releases) with [unknown] tracks to allow for these releases being combined with other sources which have the track titles and lengths.

**Comment by phw** [comment]
It is not only Tidal. Here are different tracklists (and reported release date) from the various services as of today:

**Tidal (https://tidal.com/browse/album/365033100):**
2024-06-07

2. The Bounce


**Spotify (https://open.spotify.com/intl-de/album/50D6UCzWeKbFron5Q3l7NC)**
2024-06-07

1. The Bounce


**Apple (https://music.apple.com/gb/album/the-bounce-single/1747775711)**
2024-06-21

1. The Bounce

*But the API did not return a tracklist.*

**Beatport (https://www.beatport.com/release/-/4585106)**
2024-06-07

1. The Bounce (Extended Mix)
2. The Bounce (Lampe Remix)

**Deezer (https://www.deezer.com/en/album/591275592)**
2024-06-07

1. The Bounce (Extended Mix)
2. The Bounce
3. The Bounce (Lampe Remix)

*Only track 2 and 3 are already playable, track 1 is displayed grayed out.*


My best bet on the situation is that 2024-06-21 will be the actual date of release and that this will eventually end up with a 3 track tracklist as right now only reported by Deezer. That Tidal gives the only track the number "2" strongly hints at that.

2024-06-07 is likely the date they added this to the various services, and only iTunes reports a future release date.

**Comment by kellnerd** [comment]
Related Tidal-specific issue: #65

**Comment by phw** [comment]
> My best bet on the situation is that 2024-06-21 will be the actual date of release and that this will eventually end up with a 3 track tracklist as right now only reported by Deezer.

Nope, the above assumption was wrong. The track lists are still the same as I listed above.
